Sources/Commands/SwiftTool.swift

```diff
@@ -82,10 +82,10 @@ struct TargetNotFoundDiagnostic: DiagnosticData {
 private class ToolWorkspaceDelegate: WorkspaceDelegate {
 
     /// The stream to use for reporting progress.
-    private let stdoutStream: OutputByteStream
+    private let stdoutStream: ThreadSafeOutputByteStream
 
     init(_ stdoutStream: OutputByteStream) {
-        self.stdoutStream = stdoutStream
+        self.stdoutStream = stdoutStream as? ThreadSafeOutputByteStream ?? ThreadSafeOutputByteStream(stdoutStream)
```

Comment:
I think we should handle this case in `ThreadSafeOutputByteStream`'s init.

Reply:
What's you idea? I'm not sure I follow.

Reply:
basically check if the input stream is already ThreadSafeOutputByteStream in its init so clients don't need to handle this case themselves. This is similar to AnyError: https://github.com/apple/swift-package-manager/blob/master/Sources/Basic/Result.swift#L91-L96

Reply:
But that works because `AnyError` is a struct. Won't work with classes.